Added basic homescreen and structure

pratyush-1327 opened the pull request.

pratyush-1327 committed:
Addes basic Homescreen

Added simple homescreen with a search bar and grid for cities or days

pratyush-1327 committed:
Added pubspec

Chiggy-Playz commented:
Responsive sizer is not good for making responsive designs. Look into `Breakpoint` class of [this package](https://pub.dev/packages/flutter_adaptive_scaffold) to easily handle responsive sizes. Also, we're using [GoRouter](https://pub.dev/packages/go_router) for routing pages.

pratyush-1327 committed:
Added go_router and adaptive scaffold

Added adaptive scaffold instead of responsive sizer and using go_router for routing

pratyush-1327 commented:
I have used go_routes and adpative scaffold..
Would be great if you can provide more insight about what elements to add,api's to use etc
